### PR TITLE
[cu-23v2xng][Pablo] Integrate Liquidity Bootstrapping Pool

### PR DESCRIPTION
**Pull request by vimukthi-git**

### CODE REVIEW

File: frame/pablo/src/lib.rs [hunk 1]
```diff
@@ -238,6 +262,25 @@ pub mod pallet {
 
 		#[pallet::constant]
 		type PalletId: Get<PalletId>;
+
+		// Why this is
```
Comment (hussein-aitlahcen) [inline frame/pablo/src/lib.rs:266]: Required for the spot price to be valid, as the ratio might lead to 0 before normalization

File: frame/pablo/src/liquidity_bootstrapping.rs [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+		ensure!(pool.owner == *who, Error::<T>::MustBeOwner);
+		ensure!(!base_amount.is_zero() && !quote_amount.is_zero(), Error::<T>::InvalidAmount);
+
+		// NOTE(hussein-aitlahcen): as we only allow the owner to provide liquidity, we don't
+		// mint any LP.
+		T::Assets::transfer(pool.pair.base, who, &pool_account, base_amount, keep_alive)?;
+		T::Assets::transfer(pool.pair.quote, who, &pool_account, quote_amount, keep_alive)?;
+
+		// T::deposit_event(Event::LiquidityAdded { pool_id, base_amount, quote_amount });
```
Comment (hussein-aitlahcen) [inline frame/pablo/src/liquidity_bootstrapping.rs:178]: We can remove that?

**Comment by vimukthi-git** [comment]
Need to add visualizations

File: frame/pablo/src/liquidity_bootstrapping.rs [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+					PoolConfiguration::LiquidityBootstrapping(pool.clone().value()),
+				);
+				*pool_count = pool_id.safe_add(&T::PoolId::one())?;
+				Ok(pool_id)
+			})?;
+
+		Ok(pool_id)
+	}
+
+	pub fn ensure_sale_state(
```
Comment (vimukthi-git) [inline frame/pablo/src/liquidity_bootstrapping.rs:76]:
```suggestion
	fn ensure_sale_state(
```

**Review by hussein-aitlahcen — DISMISSED** [review]

File: frame/pablo/src/liquidity_bootstrapping_tests.rs [hunk 4]
```diff
@@ -0,0 +1,672 @@
...
+		new_test_ext().execute_with(|| {
+			assert_ok!(Pablo::create(
+				Origin::signed(ALICE),
+				PoolInitConfiguration::LiquidityBootstrapping(valid_pool().value())
+			),);
+		});
+	}
+
+	// #[test]
+	// fn admin_can_create() {
```
Comment (vivekvpandya) [inline frame/pablo/src/liquidity_bootstrapping_tests.rs:141]: Is this required? if yes then may be add some comments as TODO.

Comment (vimukthi-git) [reply frame/pablo/src/liquidity_bootstrapping_tests.rs:141]: It's just a test we need to enable after `cu-23v3155`. Let me add a comment

**Review by vivekvpandya — DISMISSED** [review]

File: .github/workflows/check.yml [hunk 5]
```diff
@@ -70,7 +70,7 @@ jobs:
         run: |
           /home/runner/.cargo/bin/cargo install -f cargo-llvm-cov
```
Comment (hussein-aitlahcen) [inline .github/workflows/check.yml:71]: Here as well pin the version. :)

Comment (vimukthi-git) [reply .github/workflows/check.yml]: Done

**Review by hussein-aitlahcen — APPROVED** [review]

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-23v2xng Move liquidity bootstrapping to pablo](https://app.clickup.com/t/23v2xng)

**Review by vivekvpandya — APPROVED** [review]

**Pull request merged**